GSW/TestUtility/Source/testNS2.py

```diff
@@ -98,21 +114,31 @@ def scanPorts(): # finds and lists all available com ports
         updateLogPlain('- ' + port.description)
         foundPorts = True
     if not foundPorts:
-        updateLog('No ports detected.')
+        updateLogPlain('No ports found. Is the Teensy connected?')
                 
 # open serial port #
 def openSerialPort(event=None): # opens a new serial connection if port exists
+    if not re.match(r"COM[0-9]+", portField.get()):
```

Comment:
Was this in response to a specific error, or just checking the regex here for the hell of it?

Reply:
The `openSerialPort` command just looks for any port with a description *containing* the value in the entry field, that means that if you enter something like 'C' in the field, it will try to open every available port.

Reply:
*would try to open every available port. The regex will prevent that now.